Sources/MlemMiddleware/API Client/ApiClient.swift

```diff
@@ -109,28 +108,35 @@ public class ApiClient {
         ApiClient.apiClientCache.clean()
     }
     
-    /// Return a new `ApiClient` without a token.
-    public func loggedOut() -> ApiClient {
-        .getApiClient(for: self.baseUrl, with: nil)
+    /// Return a new guest `ApiClient`.
+    public func asGuest() -> ApiClient {
+        .getApiClient(url: self.baseUrl, username: nil)
     }
     
-    /// Return a new `ApiClient` with the given token.
-    public func loggedIn(token: String) -> ApiClient {
-        .getApiClient(for: self.baseUrl, with: token)
+    /// Return a new `ApiClient` targeting the given user.
+    public func asUser(name: String) -> ApiClient {
+        .getApiClient(url: self.baseUrl, username: name)
     }
     
     /// This should **only** be used when we get a new token for **the same** account!
     public func updateToken(_ newToken: String) {
-        guard token != nil else {
+        guard username != nil else {
             assertionFailure()
             return
         }
-        Self.apiClientCache.changeToken(for: baseUrl, oldToken: token, newToken: newToken)
         self.token = newToken
     }
     
     @discardableResult
-    func perform<Request: ApiRequest>(_ request: Request) async throws -> Request.Response {
+    func perform<Request: ApiRequest>(
+        _ request: Request,
+        allowedWhenTokenless: Bool = false // This should be `false` for the vast majority of requests, even GET requests
+    ) async throws -> Request.Response {
+        
+        guard allowedWhenTokenless || self.username == nil || self.token != nil else {
+            throw ApiClientError.noToken
+        }
```

Comment:
*nit*

Slightly nicer wording imo

```suggestion
        requiresToken: Bool = true // This should be `true` for the vast majority of requests, even GET requests
    ) async throws -> Request.Response {
        
        guard !requiresToken || self.username == nil || self.token != nil else {
            throw ApiClientError.noToken
        }
```